### PR TITLE
Circulars - Add Documentation for the Circulars Archive Search Functionality

### LINKED ISSUE #2501: Circulars - Archive Search Documentation
# Description
Add documentation on Search functionality in Circulars Archive 

# Acceptance criteria

- [ ] Describe what fields the Archive search parses
- [ ] Describe date range filter
- [ ] Describe sorting by relevance vs sorting by date
- [ ] Describe showing number of results per page

### PR DESCRIPTION
**Pull request by tylerbarna**
# Description
this adds basic search documentation for the current search paradigm. This does not make reference to lucene queries, which are still hidden behind a feature flag.

# Related Issue(s)
Resolves #2501 

# Testing
1. Navigate to http://localhost:3333/docs/circulars/archive
2. There should be a new section called "Searching the Archive"

### CODE REVIEW

**Review by dakota002 — APPROVED** [review]

**Review by lpsinger — CHANGES_REQUESTED** [review]
- No need to mention the migration here.
- No need to mention OpenSearch; that's an architecture detail that is not important for users.
- Please use the pronoun "you" to refer to the user to match the style of the rest of this document.
- Rephrase without using directions like "right" or "left" as these adjective may have no meaning for accessible modalities such as screen readers.

**Review by lpsinger — CHANGES_REQUESTED** [review]
In several places you have many words where one will do. For example:

`The Circulars archive includes a search functionality to enable you to find specific Circulars via keyword.` -> `You can search the Circulars archive by keyword.`

`This is accomplished selecting the dropdown button labeled "Filter by date".` -> `Select the dropdown button labeled "Filter by date".`

Please rephrase these two paragraphs in the plainest possible language.

File: app/routes/docs.circulars.archive.mdx [hunk 1]
```diff
@@ -13,6 +13,12 @@ Upon successful submission and distribution, a GCN Circular is assigned a number
   Go to GCN Circulars Archive
 </Link>
 
+## Searching the Archive
+
+You can search the Circulars Archive by keyword. This search tool indexes the subject, body, and submitter of all Circulars. Upon initiating a query, the search tool will return all Circulars that match the query in any of these fields. By default, results will be returned in reverse chronological order, but you can also choose to sort the results by relevance. You can navigate through the search results using pagination controls and specify the number of results to display per page. By default, 100 results are displayed per page.
```
Comment (lpsinger) [inline app/routes/docs.circulars.archive.mdx:18]:
```suggestion
You can search the Circulars Archive by keywords or phrases in the subject, body, and submitter fields. By default, results will be returned in reverse chronological order, but you can also choose to sort the results by relevance. You can navigate through the search results using pagination controls and specify the number of results to display per page. By default, 100 results are displayed per page.
```

File: app/routes/docs.circulars.archive.mdx [hunk 2]
```diff
@@ -13,6 +13,12 @@ Upon successful submission and distribution, a GCN Circular is assigned a number
...
+
+Additionally, you can filter search results by the date of their submission. Select the dropdown button labeled "Filter by date". From there, you can either select from a number of predefined time ranges (e.g., "Last Hour", "Last Day", etc.) or specify a custom date range, which can either be set as a start date, an end date, or both. All search filters can be cleared by clicking the 'x' button in the search bar.
```
Comment (lpsinger) [inline app/routes/docs.circulars.archive.mdx:20]:
```suggestion
You can also filter search results by submission date. Select the dropdown button labeled "Filter by date". From there, you can either select from a number of predefined time ranges (e.g., "Last Hour", "Last Day", etc.) or specify a custom date range, which can either be set as a start date, an end date, or both. All search filters can be cleared by clicking the 'x' button in the search bar.
```

**Review by lpsinger — CHANGES_REQUESTED** [review]

**Review by lpsinger — APPROVED** [review]

**Pull request merged**